### PR TITLE
Enhance Kueue.Queue.Status test suite

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
Previous test failure:
```bash
Test started today at 7:32 AM failed after 15m0s. ([less info](https://prow.k8s.io/spyglass/lens/metadata/iframe?req=%7B%22artifacts%22%3A%5B%22finished.json%22%2C%22podinfo.json%22%2C%22prowjob.json%22%2C%22started.json%22%5D%2C%22index%22%3A0%2C%22src%22%3A%22gs%2Fkubernetes-jenkins%2Fpr-logs%2Fpull%2Fkubernetes-sigs_kueue%2F114%2Fpull-kueue-build-image-main%2F1502261086198435840%22%7D&topURL=https%3A//prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/114/pull-kueue-build-image-main/1502261086198435840&lensIndex=0#))

Init container clonerefs not ready: (state: waiting, reason: "PodInitializing", message: "") Init container initupload not ready: (state: waiting, reason: "PodInitializing", message: "") Init container place-entrypoint not ready: (state: waiting, reason: "PodInitializing", message: "")
```

File: pkg/queue/manager.go [hunk 1]
```diff
@@ -183,13 +183,16 @@ func (m *Manager) DeleteQueue(q *kueue.Queue) {
 	delete(m.queues, key)
 }
 
-func (m *Manager) Status(q *kueue.Queue) (int, error) {
+func (m *Manager) GetPendingQW(q *kueue.Queue) (int32, error) {
```
Comment (ahg-g) [inline pkg/queue/manager.go:186]:
```suggestion
func (m *Manager) PendingWorkloads(q *kueue.Queue) (int32, error) {
```

Comment (ArangoGutierrez) [reply pkg/queue/manager.go]: sure

File: test/integration/controller/core/queue_controller_test.go [hunk 2]
```diff
@@ -79,10 +79,29 @@ var _ = ginkgo.Describe("Queue controller", func() {
...
+		admissions := []*kueue.Admission{
+			testing.MakeAdmission(clusterQueue.Name).
+				Flavor(corev1.ResourceCPU, flavorOnDemand).Obj(),
+			testing.MakeAdmission(clusterQueue.Name).
+				Flavor(corev1.ResourceCPU, flavorOnDemand).Obj(),
+			testing.MakeAdmission(clusterQueue.Name).
+				Flavor(corev1.ResourceCPU, flavorOnDemand).Obj(),
+		}
+		for i, w := range workloads {
+			w.Spec.Admission = admissions[i]
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply test/integration/controller/core/queue_controller_test.go]: cool

File: test/integration/controller/core/queue_controller_test.go [hunk 3]
```diff
@@ -79,10 +79,30 @@ var _ = ginkgo.Describe("Queue controller", func() {
 		}
 		gomega.Eventually(func() kueue.QueueStatus {
 			var updatedQueue kueue.Queue
-			gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(Queue), &updatedQueue)).To(gomega.Succeed())
+			gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(queue), &updatedQueue)).To(gomega.Succeed())
 			return updatedQueue.Status
 		}, framework.Timeout, framework.Interval).Should(testing.Equal(kueue.QueueStatus{PendingWorkloads: 3}))
 
+		ginkgo.By("Admitting workloads")
+		admissions := []*kueue.Admission{
```
Comment (ahg-g) [inline test/integration/controller/core/queue_controller_test.go:87]: remove this

Comment (ArangoGutierrez) [reply test/integration/controller/core/queue_controller_test.go]: done

File: api/v1alpha1/queue_types.go [hunk 4]
```diff
@@ -33,6 +33,7 @@ type ClusterQueueReference string
 type QueueStatus struct {
 	// PendingWorkloads is the number of workloads currently admitted to this
 	// queue not yet admitted to a ClusterQueue.
+	// +optional
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply api/v1alpha1/queue_types.go:36]: sure

**Pull request merged**